src/rdkafka.h

```diff
@@ -7085,6 +7127,46 @@ typedef enum rd_kafka_AclOperation_t {
 /**@}*/
 
 
+/**
+ * @name ListOffsets API
+ * @brief Given a topic_partition list, provides the offset information.
+ * @{
+*/
+
+/**
+ * OffsetSpec for Earliest, Latest and Timestamp of the Offset to list via ListOffsets.
+*/
+typedef enum rd_kafka_OffsetSpec_t{
+    RD_KAFKA_OFFSET_SPEC_MAX_TIMESTAMP = -3,
+    RD_KAFKA_OFFSET_SPEC_EARLIEST = -2,
+    RD_KAFKA_OFFSET_SPEC_LATEST = -1,
+}rd_kafka_OffsetSpec_t;
+
+/**
+ * Defines a ListOffsetResultInfo to be created.
+*/
+typedef struct rd_kafka_ListOffsetResultInfo_s rd_kafka_ListOffsetResultInfo_t;
+
+/**
+ * Returns the rd_kafka_topic_partition_t of the rd_kafka_ListOffsetResultInfo_t passed.
+*/
+RD_EXPORT
+const rd_kafka_topic_partition_t *rd_kafka_ListOffsetResultInfo_topic_partition(const rd_kafka_ListOffsetResultInfo_t *result_info);
+
+/**
+ * Returns the timestamp specified for the offset of the rd_kafka_ListOffsetResultInfo_t.
+*/
+RD_EXPORT
+int64_t rd_kafka_ListOffsetResultInfo_timestamp(const rd_kafka_ListOffsetResultInfo_t *result_info);
+
+/**
+ * Returns the array of pointers of rd_kafka_ListOffsetResultInfo_t given rd_kafka_ListOffsets_result_t and populates the size of the array.
+*/
+RD_EXPORT
+rd_kafka_ListOffsetResultInfo_t **rd_kafka_ListOffsets_result_infos(const rd_kafka_ListOffsets_result_t *result,size_t *cnt);
```

Comment:
Since one resultInfo holds the result of one request only it should be ListOffset only instead of ListOffsets also this was discussed in the design as well in our calls.